sherpa/astro/ui/tests/test_astro_ui.py

```diff
@@ -101,6 +120,179 @@ def test_ui_table(setup_files):
     ui.load_table(1, setup_files.fits, 4, ('R', "SUR_BRI", 'SUR_BRI_ERR'),
                   ui.Data1DInt)
 
+
+@requires_fits
+@requires_data
+def test_ui_table_noarg(setup_files, clean_astro_ui):
+    """Don't give a dataset id
+
+    It also lets us actually check the results of the load_table call
+    """
+    assert ui.list_data_ids() == []
+    ui.load_table(setup_files.fits, colkeys=['RMID', 'COUNTS'])
+    assert ui.list_data_ids() == [1]
+
+    d = ui.get_data()
+    assert d.name.endswith('1838_rprofile_rmid.fits')
+    assert isinstance(d, ui.Data1D)
+    assert len(d.x) == 38
+    assert d.x[0:5] == pytest.approx([12.5, 17.5, 22.5, 27.5, 32.5])
+    assert d.y[0:5] == pytest.approx([1529, 2014, 2385, 2158, 2013])
+
+
+def test_dataspace1d_data1dint(clean_astro_ui):
+    """Explicitly test dataspace1d for Data1DInt"""
+
+    assert ui.list_data_ids() == []
+    ui.dataspace1d(20, 30, step=2.5, id='x', dstype=ui.Data1DInt)
+
+    assert ui.list_data_ids() == ['x']
+    assert ui.get_data('x').name == 'dataspace1d'
+
+    grid = ui.get_indep('x')
+    assert len(grid) == 2
+
+    expected = numpy.asarray([20, 22.5, 25, 27.5, 30.0])
+    assert grid[0] == pytest.approx(expected[:-1])
+    assert grid[1] == pytest.approx(expected[1:])
+
+    y = ui.get_dep('x')
+    assert y == pytest.approx(numpy.zeros(4))
+
+
+def test_dataspace1d_datapha(clean_astro_ui):
+    """Explicitly test dataspace1d for DataPHA"""
+
+    assert ui.list_data_ids() == []
+
+    # Note the grid is ignored, other than the number of bins
+    ui.dataspace1d(20, 30, step=2.5, id='x', dstype=ui.DataPHA)
+
+    assert ui.list_data_ids() == ['x']
+    assert ui.get_data('x').name == 'dataspace1d'
+
+    grid = ui.get_indep('x')
+    assert len(grid) == 1
+
+    expected = numpy.asarray([1, 2, 3, 4, 5])
+    assert grid[0] == pytest.approx(expected)
+
+    y = ui.get_dep('x')
+    assert y == pytest.approx(numpy.zeros(5))
+
+    assert ui.get_exposure('x') is None
+    assert ui.get_grouping('x') is None
+    assert ui.get_quality('x') is None
+
+    assert ui.get_data('x').subtracted is False
+
+    with pytest.raises(IdentifierErr):
+        ui.get_bkg('x')
+
+
+def test_dataspace1d_datapha_bkg_nopha(clean_astro_ui):
+    """We need a PHA to create a background dataset"""
+
+    with pytest.raises(IdentifierErr) as exc:
+        ui.dataspace1d(20, 30, step=2.5, id='x', bkg_id=2, dstype=ui.DataPHA)
+
+    assert str(exc.value) == 'data set x has not been set'
+
+
+def test_dataspace1d_datapha_bkg(clean_astro_ui):
+    """Explicitly test dataspace1d for DataPHA (background)"""
+
+    # list_bkg_ids will error out until the dataset exists
+    assert ui.list_data_ids() == []
+
+    # We don't use the grid range or step size since numbins has been
+    # given.
+    ui.dataspace1d(20, 30, step=2.5, numbins=10, id='x', dstype=ui.DataPHA)
```

Comment:
Actually, since the API lets you set both I wanted an explicit test which set both (using incompatible values) to make sure the code works properly when you do set both. Alternatively we could change `dataspace1d` to error out when both are given, but that's probably excessive, and outside the scope of this PR...